src/test/skiplist_tests.cpp

```diff
@@ -20,13 +20,13 @@ BOOST_AUTO_TEST_CASE(skiplist_test)
 
     for (int i=0; i<SKIPLIST_LENGTH; i++) {
         vIndex[i].nHeight = i;
-        vIndex[i].pprev = (i == 0) ? NULL : &vIndex[i - 1];
+        vIndex[i].pprev = (i == 0) ? NULL : (vIndex.data() + i - 1);
         vIndex[i].BuildSkip();
     }
 
     for (int i=0; i<SKIPLIST_LENGTH; i++) {
         if (i > 0) {
-            BOOST_CHECK(vIndex[i].pskip == &vIndex[vIndex[i].pskip->nHeight]);
+            BOOST_CHECK(vIndex[i].pskip == vIndex.data() + vIndex[i].pskip->nHeight);
```

Comment:
@MeshCollider the advantage of `.data()` is that it is defined to return `nullptr` if `.empty()`,  compared to `&v[0]` potentially de-referencing unitialized memory.
In this event,  we assume `vIndex` is not empty, as you are de-referencing `vIndex[i]`,  therefore, it is probably not as necessary.